src/main/java/com/google/jenkins/plugins/k8sengine/client/ComputeClient.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.jenkins.plugins.k8sengine.client;
+
+import com.google.api.services.compute.Compute;
+import com.google.api.services.compute.model.Zone;
+import com.google.common.base.Preconditions;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+/**
+ * Client for communicating with the Google Compute Engine API
+ *
+ * @see <a href="https://cloud.google.com/compute/docs/reference/rest/">Compute Engine API</a>
+ */
+public class ComputeClient {
+  private final Compute compute;
+
+  /**
+   * Constructs a new {@link ComputeClient} instance.
+   *
+   * @param compute The {@link Compute} instance this class will utilize for interacting with the
+   *     GCE API.
+   */
+  public ComputeClient(Compute compute) {
+    this.compute = Preconditions.checkNotNull(compute);
+  }
+
+  /**
+   * Retrieves a list of {@link Zone} objects from the container client.
+   *
+   * @param projectId The ID of the project that zone usage is needed for.
+   * @return The retrieved list of {@link Zone} objects.
+   * @throws IOException When an error occurred attempting to get the list of zones.
+   */
+  public List<Zone> getZones(String projectId) throws IOException {
+    List<Zone> zones = compute.zones().list(projectId).execute().getItems();
+    if (zones == null) {
```

Comment:
Are we sure this is a possible case? My experience with API client libs is that they throw an exception in the error case. Also I'd be worried about this approach obfuscating an error.

Reply:
This check was based on the compute engine documentation [here](https://cloud.google.com/compute/docs/reference/rest/v1/zones/list), scroll down to examples to see the Java client library usage. While the response body itself shouldn't be null, the items could be from what this tells me. The IO exception would be thrown in "execute()" if there was a connection issue (invalid permissions, gcp is down, etc.). 

In the description of fields it doesn't make any mention about what a null result for items implies (why not a non-null but empty list instead?) so my reasoning is solely based on that fact that it was done in this example.

Reply:
Makes sense. Thanks for clarifying.